source/code/go/src/plugins/oms.go

```diff
@@ -253,6 +246,10 @@ func PostDataHelper(tailPluginRecords []map[interface{}]interface{}) int {
 			DataItems: dataItems}
 
 		marshalled, err := json.Marshal(logEntry)
+		if err != nil {
+			Log("Error while Marshalling log Entry: %s", err.Error())
+			return output.FLB_RETRY
```

Comment:
Will we end up in a situation where we will keep retrying? Previously we were sending that packet to ODS and now we are retrying from the agent.